### PR TITLE
Add literaryarts

### PR DESCRIPTION
**Pull request by brinxmat**
- Adds support for literary arts

### CODE REVIEW

File: nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/film/MovingPictureSubtypeEnum.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package no.unit.nva.model.instancetypes.artistic.film;
 
 import com.fasterxml.jackson.annotation.JsonValue;
+import no.unit.nva.model.instancetypes.artistic.WithType;
 
-public enum MovingPictureSubtypeEnum {
+public enum MovingPictureSubtypeEnum implements WithType {
```
Comment (axthosarouris) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/film/MovingPictureSubtypeEnum.java:6]: This enum returns the "type" field as its value but it lacks the JsonCreator for parsing the type.

Comment (axthosarouris) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/film/MovingPictureSubtypeEnum.java:6]: I do not think that the enum implements an interface WithType but an interface IsType.

**Review by janbj — DISMISSED** [review]

File: nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/literaryarts/LiteraryArtsSubtypeEnum.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    SHORT_STORY("ShortStory"),
+    ESSAY("Essay"),
+    TRANSLATION("Translation"),
+    RETELLING("Retelling"),
+    PLAY("Play"),
+    OTHER("Other");
+
+
+    @JsonValue
+    private final String type;
```
Comment (axthosarouris) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/literaryarts/LiteraryArtsSubtypeEnum.java:19]: Same problem with de/serializing

File: nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/film/MovingPictureSubtypeEnum.java [hunk 3]
```diff
@@ -17,6 +18,7 @@
         this.type = type;
     }
 
+    @Override
```
Comment (axthosarouris) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/film/MovingPictureSubtypeEnum.java:21]: I am not sure whether you want the subtype to be an object or a String. Now it is like this:
```
"publicationInstance" : {
        "type" : "MovingPicture",
        "subtype" : {
          "type" : "InteractiveFilm"
        },
```

If you want it to be a String i.e.,
``` 
    "publicationInstance" : {
        "type" : "MovingPicture",
        "subtype" : "InteractiveFilm"
```

then you have to add a structure similar to the following:
```
@JsonCreator
    public static MovingPictureSubtype fromString(String type){
        
    }
    
    @JsonValue
    public String toString(){
        this.getType();
    }
``` 
and remove the annotation from the field

File: nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/literaryarts/LiteraryArtsSubtype.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package no.unit.nva.model.instancetypes.artistic.literaryarts;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import nva.commons.core.JacocoGenerated;
+
+import java.util.Objects;
+
+public class LiteraryArtsSubtype {
```
Comment (axthosarouris) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/artistic/literaryarts/LiteraryArtsSubtype.java:9]: Similar problem:

```
ublicationInstance" : {
        "type" : "LiteraryArts",
        "subtype" : {
          "type" : "Poetry"
        },
```

**Pull request closed without merging**